pyanaconda/modules/storage/partitioning/interactive/change_device.py

```diff
@@ -0,0 +1,268 @@
+            return
+
+        if not self._device.raw_device.exists:
+            self._replace_device()
+        else:
+            self._change_device()
+
+    def _replace_device(self):
+        """Replace the nonexistent device with a new one.
+
+        If something has changed but the device does not exist,
+        there is no need to schedule actions on the device. It
+        is only necessary to create a new device object which
+        reflects the current choices.
+        """
+        log.debug("Replacing a nonexistent device.")
+        device = self._device
+
+        if self._should_remove_device():
+            # Remove the current device.
+            destroy_device(self._storage, device)
+
+            # We don't want to pass the device if we removed it.
+            self._request.device_spec = ""
+
+        # Create a new device.
+        log.debug("Creating a new device.")
+        arguments = get_device_factory_arguments(self._storage, self._request)
+        device = self._storage.factory_device(**arguments)
+
+        # Update the device.
+        self._device = device
+
+    def _should_remove_device(self):
+        """Should we remove the current device?"""
+        new_request = self._request
+        old_request = self._original_request
+
+        # Check the device type.
+        if old_request.device_type != new_request.device_type:
+            return True
+
+        # Check the container name.
+        if old_request.container_name and new_request.container_name != old_request.container_name:
+            return True
+
+        return False
+
+    def _change_device(self):
+        """Change the configuration of the existing device."""
+        log.debug("Modifying an existing device.")
+
+        self._revert_device_reformat()
+        self._change_device_size()
+
+        if self._should_reformat_device():
+            self._change_device_encryption()
+            self._change_device_format()
+        else:
+            self._change_device_label()
+            self._change_device_mount_point()
+
+        self._change_device_name()
+
+    def _revert_device_reformat(self):
+        """Revert reformat of the device."""
+        if self._request.reformat:
+            return
+
+        log.debug("Reverting device reformat.")
+        revert_reformat(self._storage, self._device)
+
+    def _change_device_size(self):
+        """Resize the device."""
+        size = Size(self._request.device_size)
+        original_size = Size(self._original_request.device_size)
+
+        if size == original_size:
+            return
+
+        log.debug("Changing device size: %s", size)
+        resize_device(self._storage, self._device, size, original_size)
+
+    def _should_reformat_device(self):
+        """Should we reformat the device?
+
+        It's possible that reformat is requested but the format
+        type is unchanged, in which case we're not going to schedule
+        another reformat unless encryption has changed.
```

Comment:
I'm not quite sure this is correct. We should plan an action if device reformat is requested and format does not changed, because we want to erase data. Opposite is true. We don't want to reformat if the format changed because then the reformat action is already there.

The code below seems correct to me. Do I miss something here?

Reply:
Fixed.